### PR TITLE
REF: implement ExtensionIndex._concat_same_dtype, use for IntervalIndex

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/indexes/extension.py [hunk 1]
```diff
@@ -221,6 +224,20 @@ def __getitem__(self, key):
...
+        return len(self._data)
+
+    @property
+    def size(self) -> int:
+        # If EA ever requires size, this can become self._data.size
+        return len(self)
+
+    @property
+    def dtype(self):
+        return self._data.dtype
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/extension.py]: This is already included in the base Index class (with docstring actually), so I think this is not needed here?

File: pandas/core/indexes/extension.py [hunk 2]
```diff
@@ -221,6 +224,20 @@ def __getitem__(self, key):
     def __iter__(self):
         return self._data.__iter__()
 
+    def __len__(self) -> int:
+        return len(self._data)
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/extension.py]: Same comment as below for this one

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/extension.py [hunk 3]
```diff
@@ -221,6 +224,13 @@ def __getitem__(self, key):
     def __iter__(self):
         return self._data.__iter__()
 
+    @property
+    def size(self) -> int:
+        # If EA ever requires size, this can become self._data.size
+        return len(self)
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/extension.py]: Actually, also size is already provided by the base class it seems?

**Review by jorisvandenbossche — COMMENTED** [review]
Can you maybe add a comment somewhere that those attributes (`__len__`, `dtype`) are already dispatched to `_data` in the base class?

File: pandas/core/indexes/interval.py [hunk 4]
```diff
@@ -183,23 +183,10 @@ def func(intvidx_self, other, sort=False):
 )
 @inherit_names(["set_closed", "to_tuples"], IntervalArray, wrap=True)
 @inherit_names(
-    [
-        "__len__",
```
Comment (jreback) [inline pandas/core/indexes/interval.py]: why don't we need the deleted ones?

Comment (jreback) [reply pandas/core/indexes/interval.py]: ahh i c, by reading comments.

**Review by jreback — APPROVED** [review]
lgtm.

File: pandas/core/indexes/extension.py [hunk 5]
```diff
@@ -235,6 +243,10 @@ def repeat(self, repeats, axis=None):
         result = self._data.repeat(repeats, axis=axis)
         return self._shallow_copy(result)
 
+    def _concat_same_dtype(self, to_concat, name):
```
Comment (jreback) [inline pandas/core/indexes/extension.py]: if you can type at some point (doc-string inherited)?

**Comment by jreback** [comment]
cc @jorisvandenbossche

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**